### ISSUE TITLE
filename_first doesn't work with lsp_references

### ISSUE THREAD

**Issue by ccidral** [nvim-telescope/telescope.nvim#3311]
### Description

The `filename_first` option for `path_display` doesn't seem to work with `lsp_references` as shown in this screenshot:

<img width="1512" alt="Screenshot 2024-10-01 at 5 14 34 PM" src="https://github.com/user-attachments/assets/462b2fe0-6199-4017-a420-47e042faa652">

This is my configuration:

```lua
require("telescope").setup({
  defaults = {
    path_display = { "filename_first" },
    layout_strategy = "vertical",
    layout_config = {
      height = 0.95,
      width = 0.95,
    },
  },
})
```

If I change `filename_first` to something like `shorten`, it works:

<img width="1512" alt="Screenshot 2024-10-01 at 5 21 35 PM" src="https://github.com/user-attachments/assets/eb0d4b1e-0667-4b7d-b7dd-15e87aec436f">

### Neovim version

```markdown
NVIM v0.10.1
Build type: Release
LuaJIT 2.1.1727621189
Run "nvim -V1 -v" for more info
```

### Operating system and version

macOS 14.6.1

### Telescope version / branch / rev

telescope 0.1.8

### checkhealth telescope

```markdown
==============================================================================
telescope: health#telescope#check

Checking for required plugins ~
- OK plenary installed.
- OK nvim-treesitter installed.

Checking external dependencies ~
- OK rg: found ripgrep 14.1.1
- OK fd: found fd 10.2.0

===== Installed extensions ===== ~

Telescope Extension: `fzf` ~
- OK lib working as expected
- OK file_sorter correctly configured
- OK generic_sorter correctly configured

Telescope Extension: `ui-select` ~
- No healthcheck provided
```

### Steps to reproduce

1. Set `path_display` to `filename_first`.
2. Restart Neovim.
3. Go to some source file and find references using Telescope.

### Expected behavior

Should see filename first and then directory when using `lsp_references` with Telescope.

### Actual behavior

It shows the start of the path and truncates the tail.

### Minimal config

```Lua
require("telescope").setup({
  defaults = {
    path_display = { "filename_first" },
    layout_strategy = "vertical",
    layout_config = {
      height = 0.95,
      width = 0.95,
    },
  },
})
```

**Comment by jamestrew** [comment]
This feature isn't available on 0.1.8
Only on the master branch I believe.

**Comment by ccidral** [comment]
Ok I guess I can close this issue then.
